dlio_benchmark/checkpointing/base_checkpointing.py

```diff
@@ -21,53 +21,207 @@
 from dlio_benchmark.common.enumerations import CheckpointLocationType
 from dlio_benchmark.storage.storage_factory import StorageFactory
 from dlio_benchmark.utils.config import ConfigArguments
-from dlio_benchmark.utils.utility import DLIOMPI
+from dlio_benchmark.utils.utility import DLIOMPI, utcnow
+import logging
 
+def get_datatype_size(datatype):
+    if datatype == "int8" or datatype == "uint8":
+        return 1
+    elif datatype == "fp16" or datatype == "bf16":
+        return 2
+    elif datatype == "fp32":
+        return 4
+    elif datatype == "fp64":
+        return 8
+    else:
+        raise Exception("Unsupported datatype {datatype}")
 
 class BaseCheckpointing(ABC):
 
     def __init__(self, ext):
         self.ext = ext
         self.args = ConfigArguments.get_instance()
-        checkpoint_storage = StorageFactory().get_storage(self.args.storage_type, self.args.checkpoint_folder,
+        self.checkpoint_storage = StorageFactory().get_storage(self.args.storage_type, self.args.checkpoint_folder,
                                                           self.args.framework)
-        checkpoint_storage.create_namespace(exist_ok=True)
-        rank_to_checkpoint = self.args.my_rank
-        if self.args.checkpoint_type == CheckpointLocationType.RANK_ZERO:
-            rank_to_checkpoint = 0
-        if rank_to_checkpoint == self.args.my_rank:
-            self.model_state = None
-            if self.args.model_size > 0:
-                self.model_state = {"a": self.get_tensor(self.args.model_size)}
-            self.optimization_state = None
+        self.MPI = DLIOMPI.get_instance()
+        self.comm = self.MPI.comm()
+        # define parallelism
+        self.pp = self.args.pipeline_parallelism
+        self.tp = self.args.tensor_parallelism
+        self.mp = self.tp*self.pp
+        self.dp = self.args.comm_size//self.mp
+        self.pp_rank = (self.args.my_rank // self.tp) % self.pp
+        self.tp_rank = self.args.my_rank % self.tp
+        self.dp_rank = self.args.my_rank // (self.pp*self.tp)
+        self.mp_rank = self.args.my_rank%self.mp
+        self.optimization_groups_predefined = False
+        self.layer_parameters_predefined = False
+        self.checkpoint_storage.create_namespace(exist_ok=True)
+        self.rank_to_checkpoint = self.args.my_rank
+        self.num_p = self.get_num_parameters()
+        self.checkpoint_size = 0.0
+        ss = 0.0
```

Comment:
non abbreviate

Reply:
What is ss. This was not fixed. I would recommend to just comment if you have fixed. So that I can review and resolve if it looks fine.